code/model/SiteTree.php

```diff
@@ -2643,19 +2609,18 @@ public function setMenuTitle($value) {
 	}
 	
 	/**
-	 * A flag provides the user with additional data about the current page status,
-	 * for example a "removed from draft" status. Each page can have more than one status flag.
-	 * Returns a map of a unique key to a (localized) title for the flag.
-	 * The unique key can be reused as a CSS class.
-	 * Use the 'updateStatusFlags' extension point to customize the flags.
+	 * A flag provides the user with additional data about the current page status, for example a "removed from draft"
+	 * status. Each page can have more than one status flag. Returns a map of a unique key to a (localized) title for
+	 * the flag. The unique key can be reused as a CSS class. Use the 'updateStatusFlags' extension point to customize
+	 * the flags.
 	 * 
 	 * Example (simple): 
-	 * "deletedonlive" => "Deleted"
+	 *   "deletedonlive" => "Deleted"
 	 * 
 	 * Example (with optional title attribute): 
-	 * "deletedonlive" => array('text' => "Deleted", 'title' => 'This page has been deleted')
+	 *   "deletedonlive" => array('text' => "Deleted", 'title' => 'This page has been deleted')
 	 *
-	 * @param Boolean $cached
+	 * @param bool $cached Whether to serve the fields from cache; false regenerate them
```

Comment:
> getStatusFlags is getStatusFlags(true) not getStatusFlags(null)

That would only be true if `true == null` which is not the case. Where it is called; `getStatusFlags()` is `getStatusFlags(null)`. The engine wires the default values up inside the function. I guess that's semantics though. 

The important thing is that [static analysis] engines view nullable type hints differently to nullable type hints. Could you give `null` as a value to that parameter? Then `null` is an allowed type. Whether or not it will be given a default value (inside the function) or will be coerced or whatever. The type hints (in this case the type annotations) tell a static analyser what types are allowed, not what they will become. That is why `null` should be hinted/annotated for all PHP parameters with default values.